internal/config/identity/openid/jwt.go

```diff
@@ -279,7 +279,7 @@ func parseDiscoveryDoc(u *xnet.URL, transport http.RoundTripper, closeRespFn fun
 	}
 	defer closeRespFn(resp.Body)
 	if resp.StatusCode != http.StatusOK {
-		return d, err
+		return d, errors.New(resp.Status)
```

Comment:
We should provide more context here by saying some words

```suggestion
		return d, fmt.Errorf("unexpected error returned by %s : status(%s)", u, resp.Status)  
```